x-pack/filebeat/input/httpjson/json_parser.go

```diff
@@ -0,0 +1,185 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package httpjson
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+// parse returns array of string values from JSON string
+//
+// Examples:
+// input:
+//   rawJSON={"a":[{"b":"b_value_1"},{"b":"b_value_2"},{"b":"b_value_3"}]}
+//   key=a.#.b
+// output:
+//   ["a_value_1", "a_value_2"]
+// input:
+//   rawJSON=[{"a":"a_value_1"},{"a":"a_value_2"}]
+//   key=#.a
+// output:
+//   ["a_value_1", "a_value_2"]
+func parse(rawJSON, key string) ([]string, error) {
```

Comment:
Make `rawJSON` a `[]byte` since the call site at L327 converts a `[]byte` to a string to satisfy this signature, resulting in an unnecessary allocation.

Suggest rename `rawJSON` to something like `data` and function that more clearly reflects its function; this is not a parser (same below). `getKeyedArrayValue`? to mirror the naming below.

Reply:
Got it. will update it.